pkg/binlog/binlog.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binlog
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/pingcap/errors"
+)
+
+// Client is the client of binlog.
+type Client struct {
+	tls        *tls.Config
+	httpClient *http.Client
```

Comment:
We have to close the http client if TLS is enabled.

Reply:
https://github.com/pingcap/tidb-operator/pull/3973/commits/a130b0fcf2d058c76dbf0380571df3e4bb415e0b